DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/TokenCheckMiddleware.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Net;
+using System.Threading.Tasks;
+
+using GitHubExtension.Infrastructure.Constants;
+using GitHubExtension.Infrastructure.Extensions.Owin;
+
+using Microsoft.Owin;
+
+namespace GitHubExtension.EntryPoint
+{
+    /// <summary>
+    /// Middleware that check external access token presence in claims
+    /// </summary>
+    public class TokenCheckMiddleware : OwinMiddleware
+    {
+        private const string NoToken = "You don't have access token to enter this resource";
+
+        public TokenCheckMiddleware(OwinMiddleware next)
+            : base(next)
+        {
+        }
+
+        public override async Task Invoke(IOwinContext context)
+        {
+            if (context.IsAuthenticated())
+            {
+                if (!context.Authentication.User.HasClaim(c => c.Type == ClaimConstants.ExternalAccessToken))
+                {
+                    context.Authentication.SignOut();
```

Comment:
Or maybe it's better to write authentication middleware to replace github middleware